cedar-policy-core/src/parser.rs

```diff
@@ -47,6 +47,28 @@ pub fn parse_policyset(text: &str) -> Result<ast::PolicySet, Vec<err::ParseError
         .ok_or(errs)
 }
 
+/// Like `parse_policyset()`, but also returns the (lossless) original text of
+/// each individual policy.
+pub fn parse_policyset_and_also_return_policy_text(
+    text: &str,
+) -> Result<(HashMap<ast::PolicyID, &str>, ast::PolicySet), err::ParseErrors> {
+    let mut errs = Vec::new();
+    let cst = text_to_cst::parse_policies(text).map_err(err::ParseErrors)?;
+    let pset = cst
+        .to_policyset(&mut errs)
+        .ok_or_else(|| err::ParseErrors(errs.clone()))?;
+    let texts = cst
+        .with_generated_policyids()
+        .expect("shouldn't be None since parse_policies() and to_policyset() didn't return Err")
```

Comment:
Are we sure on this?

Reply:
This `expect` is also present in the function below, with the `allow(clippy::expect_used)` annotation. If we're confident there, I'm confident here